### PR TITLE
update CareLink deviceMeta status reasons to accord with revised data model

### PR DESCRIPTION
**Pull request by jebeck**
Dependency hell: required jellyfish v0.8.0 for Insulet demo broke CareLink. This fixes (although it also required additional tweaks to jellyfish: https://github.com/tidepool-org/jellyfish/pull/54)

This is already merged into the branch I'll be building for the devel Chrome store, and there will be a few small branches like that this weekend. I will be creating a list (incl. order in which I think they should be reviewed) by end of weekend.

### CODE REVIEW

**Comment by jebeck** [comment]
Hey @jh-bate now that jellyfish v0.8.1 is deployed on devel and staging, master of the uploader will be broken for CareLink until this is merged. Can you take a look at it tonight?

File: lib/carelink/carelinkSimulator.js [hunk 1]
```diff
@@ -312,7 +312,7 @@ exports.make = function(config){
        * during an LGS. We are already handling these in our basal clock, so we just
        * push the clock forward and exit early when we come across one.
        */
-      if (currStatus != null && currStatus.status === 'suspended' && currStatus.reason === 'low_glucose') {
+      if (currStatus != null && currStatus.status === 'suspended' && (currStatus.payload && currStatus.payload.cause === 'low_glucose')) {
```
Comment (jh-bate) [inline lib/carelink/carelinkSimulator.js]: seems an awful one liner check but prob good enough for now

File: lib/carelink/carelinkSimulator.js [hunk 2]
```diff
@@ -861,7 +867,7 @@ exports.make = function(config){
       var event = ensureTimestamp(combineArguments(arguments));
       // threshold suspends involve several (usually 3-4) suspend events in succession
       // we only care about the first, so if the currStatus is already `suspended`, we return early
-      if (currStatus != null && currStatus.status === 'suspended' && event.reason === 'low_glucose') {
+      if (currStatus != null && currStatus.status === 'suspended' && (event.payload && event.payload.cause === 'low_glucose')) {
```
Comment (jh-bate) [inline lib/carelink/carelinkSimulator.js]: ditto on the one liner check

File: lib/carelink/suspend.js [hunk 3]
```diff
@@ -33,23 +33,68 @@ module.exports = function (timezone) {
...
-                return { status: 'suspended', reason: 'low_glucose' };
-
+                return {
+                  status: 'suspended',
+                  reason: {
+                    suspended: 'automatic'
+                  },
+                  payload: {
+                    cause: 'low_glucose',
+                    code: reason
```
Comment (jh-bate) [inline lib/carelink/suspend.js]: so the code references the reason above?

**Comment by jh-bate** [comment]
made some comments will chat asap

File: lib/carelink/suspend.js (hunk 3 above)
Comment (jebeck) [inline lib/carelink/suspend.js]: `reason` is the raw CareLink string, so I named the attr `code`. Does that make it clearer?

**Pull request merged**